client/checkout/utils/upe.js

```diff
@@ -331,26 +337,106 @@ export const togglePaymentMethodForCountry = ( upeElement ) => {
 	const supportedCountries =
 		paymentMethodsConfig[ paymentMethodType ].countries;
 	const selectedPaymentMethod = getSelectedUPEGatewayPaymentMethod();
+	// Simplified approach - find the form ancestor and then search within it
+	let billingInput = upeElement
+		?.closest( 'form.checkout, form#add_payment_method' )
+		?.querySelector( '#billing_country' );
+
+	// If not found, fallback to the search in the whole document
+	if ( ! billingInput ) {
+		billingInput = document.querySelector( '#billing_country' );
+	}
 
 	/* global wcpayCustomerData */
 	// in the case of "pay for order", there is no "billing country" input, so we need to rely on backend data.
 	const billingCountry =
-		document.getElementById( 'billing_country' )?.value ||
-		wcpayCustomerData?.billing_country ||
-		'';
+		billingInput?.value || wcpayCustomerData?.billing_country || '';
 
-	const upeContainer = document.querySelector(
-		'.payment_method_woocommerce_payments_' + paymentMethodType
-	);
+	const upeContainer = upeElement?.closest( '.wc_payment_method' );
 	if ( supportedCountries.includes( billingCountry ) ) {
 		upeContainer.style.removeProperty( 'display' );
 	} else {
 		upeContainer.style.display = 'none';
-		// if the toggled off payment method was selected, we need to fall back to credit card
 		if ( paymentMethodType === selectedPaymentMethod ) {
-			document
-				.querySelector( '#payment_method_woocommerce_payments' )
-				.click();
+			const defaultPaymentMethod = 'card';
+			const cardPaymentForm = document.querySelector(
+				`.wcpay-upe-form[data-payment-method-type="${ defaultPaymentMethod }"]`
+			);
+
+			const cardPaymentMethodInput = cardPaymentForm
+				?.closest( '.wc_payment_method' )
```

Comment:
unfortunately, some mention of WC elements directly here, because we want to click on those elements